core/modules/node/tests/node.test

```diff
@@ -887,6 +890,70 @@ class NodeLayoutPreviewTestCase extends BackdropWebTestCase {
   }
 }
 
+/**
+ * Tests that the correct layout is used for node revisions.
+ */
+class NodeLayoutRevisionTestCase extends BackdropWebTestCase {
+
+  /**
+   * Sets up test case.
+   */
+  public function setUp() {
+    parent::setUp('node_layout_preview_test');
+
+    // Create and login user.
+    $web_user = $this->backdropCreateUser(array(
+      'view revisions',
+    ));
+    $this->backdropLogin($web_user);
+    $nodes = array();
+
+    // Create initial node.
+    $node = $this->backdropCreateNode();
+    $settings = get_object_vars($node);
+    $settings['revision'] = 1;
+    $settings['is_active_revision'] = TRUE;
+    $nodes[] = $node;
+
+    // Create a revision with random title and body and update variables.
+    $this->backdropCreateNode($settings);
+    // Make sure we get revision information.
+    $node = node_load($node->nid);
+    $settings = get_object_vars($node);
+    $settings['is_active_revision'] = TRUE;
+    $nodes[] = $node;
+
+    $this->nodes = $nodes;
+  }
+
+  /**
+   * Tests that the correct layout is used for node revisions.
+   */
+  public function testNodeLayoutRevisions() {
+    $nodes = $this->nodes;
+
+    // Confirm that our layout shows up on both the original node and its
+    // revision.
+    foreach ($nodes as $node) {
+      $node = $nodes[0];
```

Comment:
This line should be removed, as we are already iterating through the 2 node revisions:
```suggestion
```